### PR TITLE
Use standard Nullable

### PR DESCRIPTION
**Pull request by lptr**
Replace our custom `org.gradle.api.Nullable` with the standard `javax.annotation.Nullable` from JSR-305.

### CODE REVIEW

**Comment by bamboo** [comment]
👍 

It would be nice to standardise on `javax.annotation.Nonnull` as well.

**Comment by lptr** [comment]
@bamboo I'm not aware of any use of a `@Nonnull` annotation. Where did you see that?

**Comment by bamboo** [comment]
@lptr I mean, it would be nice to start using `Nonnull` as well.

**Comment by lptr** [comment]
Understood. Would you mind doing the review?

I'm running a Stage 2 build here: https://builds.gradle.org/viewQueued.html?itemId=4806061&tab=queuedBuildOverviewTab

File: subprojects/platform-base/src/main/java/org/gradle/api/BuildableComponentSpec.java [hunk 1]
```diff
@@ -18,6 +18,8 @@
 
 import org.gradle.platform.base.ComponentSpec;
 
+import javax.annotation.Nullable;
```
Comment (lptr) [inline subprojects/platform-base/src/main/java/org/gradle/api/BuildableComponentSpec.java]: No, imports take precedence. That's what IntelliJ is reporting, and I also ran this test:

```groovy
    def "test"() {
        def thing = BuildableComponentSpec.declaredMethods.find { it.name == "getBuildTask" }
        expect:
        thing.annotations.each {
            println "$thing has annotation: $it"
        }
    }
```

...which printed:

```text
public abstract org.gradle.api.Task org.gradle.api.BuildableComponentSpec.getBuildTask() has annotation: @javax.annotation.Nullable()
```

**Pull request merged**